common/flatpak-run-wayland.c

```diff
@@ -217,5 +292,17 @@ flatpak_run_add_wayland_args (FlatpakBwrap *bwrap,
                               NULL);
       flatpak_bwrap_add_runtime_dir_member (bwrap, wayland_display);
     }
+
+  if (!inherit_wayland_socket)
+    {
+      flatpak_bwrap_unset_env (bwrap, "WAYALND_SOCKET");
+      /* In order avoid thread-safety issue of setenv/unsetenv, set FD_CLOEXEC on
+       * it instead of close it so the fd won't be leak to application. Otherwise
+       * file descriptor reuse and multi threading may potentially make us close
+       * the wrong file descriptor. */
```

Comment:
```suggestion
      /* Make sure the fd is close-on-execute so it won't be inherited by the
       * application. We do this in preference to closing it, because if this function
       * was somehow called twice, and the same fd number was reused for an
       * unrelated purpose, we don't want to close the unrelated fd the
       * second time. */
```

Reply:
applied